cassandra-adapter/src/main/java/org/polypheny/db/adapter/cassandra/CassandraStore.java

```diff
@@ -129,9 +126,40 @@ public CassandraStore( int storeId, String uniqueName, Map<String, String> setti
             this.dbHostname = this.embeddedCassandra.getAddress().getHostAddress();
             this.dbPort = this.embeddedCassandra.getPort();
 
+            this.dbKeyspace = "keyspace";
+            this.dbUsername = "cassandra";
+            this.dbPassword = "";
             log.warn( "Embedded cassandra address: {}:{}", this.dbHostname, this.dbPort );
-        } else {
+        } else if ( deployMode == DeployMode.DOCKER ) {
+            this.dbUsername = "cassandra";
+            this.dbPassword = "cassandra";
+
+            DockerManager.Container container = new ContainerBuilder( getAdapterId(), "bitnami/cassandra:3", getUniqueName(), Integer.parseInt( settings.get( "instanceId" ) ) )
+                    .withMappedPort( 9042, Integer.parseInt( settings.get( "port" ) ) )
+                    //.withEnvironmentVariables( Arrays.asList( "CASSANDRA_USER=" + this.dbUsername, "CASSANDRA_PASSWORD=" + this.dbPassword ) )
+                    .build();
+            DockerManager.getInstance().initialize( container ).start();
+
+            // TODO: Wait until ready
+            try {
+                TimeUnit.SECONDS.sleep( 20 );
+            } catch ( InterruptedException e ) {
+                // ignore
+            }
```

Comment:
We need a better solution for this.

Reply:
see https://github.com/polypheny/Polypheny-DB/pull/314#discussion_r618721706